### PR TITLE
Remove docs dependency on setuptools_scm

### PR DESCRIPTION
**Pull request by cidrblock**
Part of ongoing work to remove `ignore_imports` for mypy.

In this case, rather than account for setuptools_scm in the mypy.ini as untyped (https://github.com/pypa/setuptools_scm/issues/501), simply remove the dependency on it entirely.

### CODE REVIEW

**Review by webknjaz — CHANGES_REQUESTED** [review]
setuptools-scm is necessary for building from Git checkout. It is by design that it produces unique versions for builds that make it easier to identify the source of a specific build. It should remain in use.

It is also visible @ https://ansible-navigator--989.org.readthedocs.build/en/989/changelog/#v1-1-0-alpha-1-as-of-today-unreleased-draft that this PR breaks both the displayed version and the hardcoded version identifier that became incompatible with PEP 440 (it must be compliant for PyPI uploads not to be rejected and for pip to understand the semantics of the given version).

File: docs/conf.py [hunk 1]
```diff
@@ -36,23 +34,19 @@
 author = f"{project} project contributors"
 copyright = author  # pylint:disable=redefined-builtin
 
-# fmt: off
+NAVIGATOR_VERSION = version("ansible_navigator")
+
 # The short X.Y version
-version = ".".join(
```
Comment (webknjaz) [inline docs/conf.py:41]: This is a Sphinx setting. Don't lose it. Although, I'm still against breaking what setuptools-scm provides. If you want to reuse the installed dist metadata, you'd have to (1) integrate setuptools-scm into packaging first and then (2) somehow solve the self-reinstallation on every local build in an optimal way (without `tox -r`).

File: docs/conf.py [hunk 2]
```diff
@@ -3,17 +3,15 @@
 # Ref: https://www.sphinx-doc.org/en/master/usage/configuration.html
 """Configuration file for the Sphinx docs."""
 
-from functools import partial
+
+from importlib.metadata import version
```
Comment (webknjaz) [inline docs/conf.py:7]: This shadows a built-in/internal Sphinx setting, rename it.

File: docs/conf.py [hunk 3]
```diff
@@ -193,7 +187,7 @@
     "github_user": github_repo_org,
     "github_repo": github_repo_name,
     "github_version": "main/docs/",
-    "current_version": version,
```
Comment (webknjaz) [inline docs/conf.py:196]: You don't need to do this, you still have to resurrect the `version` variable as a string, since it's something that Sphinx uses internally and exposes as a `|version|` substitution.

File: docs/conf.py [hunk 4]
```diff
@@ -36,23 +34,19 @@
...
-# fmt: off
-# The short X.Y version
-version = ".".join(
-    get_scm_version(
-        local_scheme="no-local-version",
-    ).split(".")[:3],
-)
+NAVIGATOR_VERSION = get_version("ansible_navigator")
+
+# The short X.Y.Z version
```
[the remark this answers is not in the document]
Comment (cidrblock) [reply docs/conf.py:39]: but it's totally wrong? :)

[the remark this answers is not in the document]
Comment (webknjaz) [reply docs/conf.py:39]: It's sort of a balance. I'd maybe put an additional inline comment to explain the rest?

File: pyproject.toml [hunk 5]
```diff
@@ -45,6 +45,8 @@
     name = "Contributor-facing changes"
     showcontent = true
 
+[build-system]
+requires = ["setuptools>=45", "wheel", "setuptools_scm[toml]>=6.2"]
```
Comment (webknjaz) [inline pyproject.toml:49]: `wheel` shouldn't be in this list, `setuptools`' PEP 517 build backend auto-adds it already.

File: pyproject.toml [hunk 6]
```diff
@@ -95,3 +97,5 @@ skip_glob = ["tests/fixtures/common/collections*"] # Skip ansible content due to
 
     [tool.pylint.format]
     max-line-length = 100
+
+[tool.setuptools_scm]
```
Comment (webknjaz) [inline pyproject.toml:101]: This whole dist packaging setup refactoring should go into a separate PR as it's not related to docs. It could be accepted separately while the rest of this PR is still in a state that has to be rejected.

File: docs/conf.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (cidrblock) [reply docs/conf.py:39]: I'm assuming you mean this:  https://github.com/sphinx-doc/sphinx/blob/a8eb1aab72a514b7257c780c0c57432732a702fb/sphinx/templates/quickstart/conf.py_t

I'm going to leave this, for exactly the reason you mentioned, when diffed, it will highlight the the deviation from the tempalte.

File: pyproject.toml (hunk 5 above)
Comment (webknjaz) [reply pyproject.toml:49]: (and of course move this to a separate PR as well)

File: pyproject.toml (hunk 6 above)
Comment (cidrblock) [reply pyproject.toml:101]: It can't yet.  The _version file contains more than just the package version.

File: docs/conf.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (cidrblock) [reply docs/conf.py:39]: I'll compromise, brb

File: pyproject.toml (hunk 5 above)
Comment (webknjaz) [reply pyproject.toml:49]: See also: https://github.com/ansible/ansible-navigator/issues/554

File: pyproject.toml (hunk 6 above)
Comment (webknjaz) [reply pyproject.toml:101]: No, you can do that now. Just don't point it at an existing file. Instead, point it at a gitignored `_scm_version.py` that you'd import from the `_version.py`.
Example: https://github.com/sphinx-contrib/sphinxcontrib-towncrier/blob/532e343/pyproject.toml#L16 + https://github.com/sphinx-contrib/sphinxcontrib-towncrier/blob/532e343/src/sphinxcontrib/towncrier/_version.py.

P.S. Also, the same approach is used in `ansible-pylibssh`.

Comment (webknjaz) [reply pyproject.toml:101]: Pro tip: create a [`_scm_version.pyi`](https://github.com/sphinx-contrib/sphinxcontrib-towncrier/blob/532e343/src/sphinxcontrib/towncrier/_scm_version.pyi) to take care of typing.

**Comment by cidrblock** [comment]
This PR is being clsoed.
